packages/forms/src/components/DateInput/DateInput.stories.tsx

```diff
@@ -0,0 +1,231 @@
+  defaultValue?: string;
+};
+
+const Template: ComponentStory<DefaultValueDateInputArgs> = (args) => {
+  const { defaultValue, ...rest } = args;
+  return (
+    <Form
+      {...(defaultValue && {
+        options: {
+          defaultValues: {
+            [rest.name]: defaultValue,
+          },
+        },
+      })}
+      onSubmit={(data) => action("Submit Form")(data)}
+    >
+      <DateInput {...rest} />
+      <Submit />
+    </Form>
+  );
+};
+
+export const Default = Template.bind({});
+
+export const WithDefaultValue = Template.bind({});
+WithDefaultValue.args = {
+  defaultValue: "2023-02-03",
+};
+
+export const Required = Template.bind({});
+Required.args = {
+  rules: {
+    required: "This field is required",
+  },
+};
+
+export const WithinRange = Template.bind({});
+WithinRange.args = {
+  rules: {
+    min: {
+      value: "2023-01-01",
+      message: "Must be after 2023-01-01",
+    },
+    max: {
+      value: "2024-01-01",
+      message: "Must be before 2024-01-01",
+    },
+  },
+};
+
+export const OnlyYearAndMonth = Template.bind({});
+OnlyYearAndMonth.args = {
+  show: [DATE_SEGMENT.Year, DATE_SEGMENT.Month],
+};
+
+const ValidationDependantInputs = ({
+  name,
+  id,
+  legend,
+  ...rest
+}: DateInputProps) => {
+  const watchFirstInput = useWatch({ name });
+
+  return (
+    <>
+      <DateInput {...{ name, id, legend }} {...rest} />
+      <DateInput
+        name={`${name}Two`}
+        id={`${id}Two`}
+        legend={`${legend} Two`}
+        rules={{
+          min: {
+            value: watchFirstInput,
+            message: `Must be after ${watchFirstInput}`,
+          },
+        }}
+      />
+    </>
+  );
+};
+
+const ValidationDependantTemplate: ComponentStory<DateInputArgs> = (args) => {
+  return (
+    <Form
+      options={{ mode: "onChange" }}
+      onSubmit={(data) => action("Submit Form")(data)}
+    >
+      <ValidationDependantInputs {...args} />
+      <Submit />
+    </Form>
+  );
+};
+
+export const SecondComesAfterFirst = ValidationDependantTemplate.bind({});
+
+const RenderDependantInput = ({ name }: Pick<DateInputProps, "name">) => {
+  const watchFirstInput = useWatch({ name });
+  const inputDate = watchFirstInput
+    ? formDateStringToDate(watchFirstInput)
+    : null;
+
+  return inputDate && isBefore(new Date(), inputDate) ? (
+    <Input type="text" id="signature" name="signature" label="Signature" />
+  ) : (
+    <p data-h2-margin="base(x1, 0)">
+      Please select a date in the past to continue.
+    </p>
+  );
+};
+
+const RenderDependantTemplate: ComponentStory<DateInputArgs> = (args) => {
+  const { name, ...rest } = args;
+  return (
+    <Form
+      options={{ mode: "onChange" }}
+      onSubmit={(data) => action("Submit Form")(data)}
+    >
+      <DateInput name={name} {...rest} />
+      <RenderDependantInput name={name} />
+      <Submit />
+    </Form>
+  );
+};
+
+export const HideInputWhenInvalid = RenderDependantTemplate.bind({});
```

Comment:
The conditional visiblity seems a little glitchy on this one.

https://user-images.githubusercontent.com/8978655/225688735-e8bb2312-4202-496d-bcd8-d9b827ab2ac6.mp4

Reply:
Oops, I was using the wrong condition 🤦‍♀️ Fixed in [update to on submit, fix conditional render story](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5928/commits/6638dd6b46366b817ea3d488aaab3801773e63bb)